superset/daos/tag.py

```diff
@@ -394,7 +394,11 @@ def create_tag_relationship(
         updated_tagged_objects = {
             (to_object_type(obj[0]), obj[1]) for obj in objects_to_tag
         }
-        tagged_objects_to_delete = current_tagged_objects - updated_tagged_objects
+
+        if not objects_to_tag:
+            tagged_objects_to_delete = current_tagged_objects
+        else:
+            tagged_objects_to_delete = current_tagged_objects - updated_tagged_objects
```

Comment:
maybe simplify with a ternary? Something like:
`tagged_objects_to_delete = current_tagged_objects - updated_tagged_objects if objects_to_tag else current_tagged_objects`
just slightly easier to read imo

Reply:
I think you can replace all of this with:

```python
tagged_objects_to_delete = current_tagged_objects - updated_tagged_objects
```

Since if `objects_to_tag` is falsy then `updated_tagged_objects` is an empty set.